src/t_stream.c

```diff
@@ -2029,30 +2029,56 @@ void xackCommand(client *c) {
     if (ids != static_ids) zfree(ids);
 }
 
-/* XPENDING <key> <group> [<start> <stop> <count> [<consumer>]]
+/* XPENDING <key> <group> [<start> <stop> <count> [<consumer>] [IDLE <idle>]]
  *
  * If start and stop are omitted, the command just outputs information about
  * the amount of pending messages for the key/group pair, together with
  * the minimum and maximum ID of pending messages.
  *
  * If start and stop are provided instead, the pending messages are returned
- * with informations about the current owner, number of deliveries and last
+ * with information about the current owner, number of deliveries and last
  * delivery time and so forth. */
 void xpendingCommand(client *c) {
     int justinfo = c->argc == 3; /* Without the range just outputs general
                                     informations about the PEL. */
     robj *key = c->argv[1];
     robj *groupname = c->argv[2];
-    robj *consumername = (c->argc == 7) ? c->argv[6] : NULL;
+    robj *consumername = NULL;
     streamID startid, endid;
     long long count;
+    long long minidle = 0;
+
+    /* Must contain at least key and group. */
+    if (c->argc < 3) {
+        addReply(c,shared.syntaxerr);
+        return;
+    }
```

Comment:
i see the arity in the command tale is -3, so this isn't needed.
one can say that it makes the code more readable and safe.
but the entire code base is built with that assumption.